### ISSUE TITLE
Two issues with hexadecimal floats

### ISSUE THREAD

**Issue by Hibou57** [githwxi/ATS-Postiats#203]
I suspect the two issues are related.

First, a working example which compiles and run:

        val v1 = 0xF.1P0
        val v2 = 15.0625
        //
        implement main0() = begin
           println! v1;
           println! v2;
        end

If the exponent part of the hexadecimal float `v1`, is removed, I get a compile‑time error, but no parsing nor type‑check error. The compile‑time error says an hexadecimal float requires an exponent part.

The second, case is if If do `val v1:float = 0xF.1f` then it does not type‑check; seems `0xF.1f` is seen as a `double` instead of a `float` although there is an "F" suffix. The type‑checking error disappears if there is an exponent part. Ex, this type‑checks: `val v1:float = 0xF.1P0f`.

I first encountered the second case and wanted to fix it. But then I encountered the first case and wondered if an exponent is required or not. This is strange it triggers a compile‑time error but no parsing error.

What is the intent? Should I try to fix `pats_lexing.dats`? But may be the compile‑time error will remain …

**Comment by githwxi** [comment]
I suppose I didn't read the specification very carefully.
I suspected that an exponent part is always needed when
forming a hexidecimal float. Maybe should just issue a message
telling the user that an exponent is needed in such cases.

**Comment by Hibou57** [comment]
You are right. The C standard says the exponent part is not optional with hexadecimal float constant. It’s at section 6.4.4.2 here: http://www.open-std.org/jtc1/sc22/WG14/www/docs/n1256.pdf

So yes, telling sooner is the best to do. Will have a look at it tomorrow.

**Comment by Hibou57** [comment]
Oops, another one: `0x.P0` should be rejected too. At least one of the integer or fractional part is required. Will fix this one later.
